### ISSUE TITLE
Witcher 3 ForceFeedback/vibrations doesnt work.

### ISSUE THREAD

**Issue by rofled69** [x360ce/x360ce#98]
FF doesnt work, tried many settings, force passthrough etc but no go.
My gamepad is x360 wired but it probabbly doesnt matter.
Same thing is with GTA5 but only when game is minimized and restored, FF stops working.
Almost all 64bit games have problems with FF, ive noticed.
Log looks like this after using ingame skill where shoud vibrate:
[TIME]      [THREAD]    [LOG]
18:52:02.602    00006020    InputHook starting...
18:52:02.602    00006020    HookMask 0x00000002: HOOK_COM
18:52:02.602    00006020    Hooking COM
18:52:02.603    00006020    Hook for CoCreateInstance successed
18:52:02.603    00006020    Hook for CoCreateInstanceEx successed
18:52:02.603    00006020    Hook for CoGetClassObject successed
18:52:02.603    00006020    Hook for CoUninitialize successed
18:52:03.564    00006020    CoCreateInstance {5A508685-A254-4FBA-9B82-9A24B00306AF} => {8BCF1F58-9FE7-4583-8AC6-E2ADC465C8BB}
18:52:03.564    00006500    Waiting 45s for hooks...
18:52:03.565    00006020    CoCreateInstance {BCDE0395-E52F-467C-8E3D-C4579291692E} => {A95664D2-9614-4F35-A746-DE8DB63617E6}
18:52:03.566    00006020    CoCreateInstance {BCDE0395-E52F-467C-8E3D-C4579291692E} => {A95664D2-9614-4F35-A746-DE8DB63617E6}
18:52:03.568    00006020    CoCreateInstance {BCDE0395-E52F-467C-8E3D-C4579291692E} => {A95664D2-9614-4F35-A746-DE8DB63617E6}
18:52:03.570    00006020    CoCreateInstance {BCDE0395-E52F-467C-8E3D-C4579291692E} => {A95664D2-9614-4F35-A746-DE8DB63617E6}
18:52:03.572    00006020    CoCreateInstance {BCDE0395-E52F-467C-8E3D-C4579291692E} => {A95664D2-9614-4F35-A746-DE8DB63617E6}
18:52:03.583    00006020    CoCreateInstance {BCDE0395-E52F-467C-8E3D-C4579291692E} => {A95664D2-9614-4F35-A746-DE8DB63617E6}
18:52:03.584    00006020    CoCreateInstance {BCDE0395-E52F-467C-8E3D-C4579291692E} => {A95664D2-9614-4F35-A746-DE8DB63617E6}
18:52:06.747    00006020    CoCreateInstance {5A508685-A254-4FBA-9B82-9A24B00306AF} => {8BCF1F58-9FE7-4583-8AC6-E2ADC465C8BB}
18:52:06.747    00006020    CoCreateInstance {BCDE0395-E52F-467C-8E3D-C4579291692E} => {A95664D2-9614-4F35-A746-DE8DB63617E6}
18:52:06.748    00006020    CoCreateInstance {BCDE0395-E52F-467C-8E3D-C4579291692E} => {A95664D2-9614-4F35-A746-DE8DB63617E6}
18:52:06.750    00006020    CoCreateInstance {BCDE0395-E52F-467C-8E3D-C4579291692E} => {A95664D2-9614-4F35-A746-DE8DB63617E6}
18:52:06.752    00006020    CoCreateInstance {BCDE0395-E52F-467C-8E3D-C4579291692E} => {A95664D2-9614-4F35-A746-DE8DB63617E6}
18:52:06.754    00006020    CoCreateInstance {BCDE0395-E52F-467C-8E3D-C4579291692E} => {A95664D2-9614-4F35-A746-DE8DB63617E6}
18:52:06.783    00006020    x360ce (x64) 3.4.1.1299 started for "witcher3.exe"
18:52:06.783    00006020    OS: "Microsoft Windows 8.1 Pro (x64)"
18:52:06.783    00006020    Using config file:
18:52:06.783    00006020    G:\GOG Games\The Witcher 3 Wild Hunt\bin\x64\x360ce.ini
18:52:06.784    00006020    WARNING: Configuration file version mismatch detected
18:52:06.784    00006020    Mapping could not be determined for IG_ad61b5c05a2111e48007444553540000 button D-pad Up
18:52:06.784    00006020    Mapping could not be determined for IG_ad61b5c05a2111e48007444553540000 button D-pad Down
18:52:06.784    00006020    Mapping could not be determined for IG_ad61b5c05a2111e48007444553540000 button D-pad Left
18:52:06.784    00006020    Mapping could not be determined for IG_ad61b5c05a2111e48007444553540000 button D-pad Right
18:52:06.785    00006020    [PAD1] Device created
18:52:06.786    00006020    [PAD1] Detected axis count: 6
18:52:06.788    00006020    [DINPUT] [PAD1] IsForceSupported: 0x0000170F YES
18:52:06.789    00006020    ForceFeedback effect 'Constant'. IsConstant = 1, IsPeriodic = 0
18:52:06.789    00006020    ForceFeedback effect 'Ramp'. IsConstant = 0, IsPeriodic = 0
18:52:06.789    00006020    ForceFeedback effect 'Square Wave'. IsConstant = 0, IsPeriodic = 1
18:52:06.789    00006020    ForceFeedback effect 'Sine Wave'. IsConstant = 0, IsPeriodic = 1
18:52:06.790    00006020    ForceFeedback effect 'Triangle Wave'. IsConstant = 0, IsPeriodic = 1
18:52:06.790    00006020    ForceFeedback effect 'Sawtooth Up Wave'. IsConstant = 0, IsPeriodic = 1
18:52:06.791    00006020    ForceFeedback effect 'Sawtooth Down Wave'. IsConstant = 0, IsPeriodic = 1
18:52:06.791    00006020    ForceFeedback effect 'Spring'. IsConstant = 0, IsPeriodic = 0
18:52:06.792    00006020    ForceFeedback effect 'Damper'. IsConstant = 0, IsPeriodic = 0
18:52:06.792    00006020    ForceFeedback effect 'Inertia'. IsConstant = 0, IsPeriodic = 0
18:52:06.792    00006020    ForceFeedback effect 'Friction'. IsConstant = 0, IsPeriodic = 0
18:52:06.793    00006020    ForceFeedback effect 'Custom'. IsConstant = 0, IsPeriodic = 0
18:52:06.793    00006020    [PAD1] Initialized for user 0
18:52:28.507    00006020    Type 0 Axes 2 OMag 10000 LSpeed 32767 RSpeed 0 LMag 5000 RMag 0 LPeriod 60000 RPeriod
18:52:28.509    00006020    [PAD1] CreateEffect succeded effectIndex 1
18:52:28.509    00006020    [PAD1] CreateEffect succeded effectIndex 1
18:52:28.598    00006020    Type 0 Axes 2 OMag 10000 LSpeed 45874 RSpeed 0 LMag 7000 RMag 0 LPeriod 60000 RPeriod
18:52:28.716    00006020    Type 0 Axes 2 OMag 10000 LSpeed 13107 RSpeed 0 LMag 2000 RMag 0 LPeriod 60000 RPeriod
18:52:28.740    00006020    Type 0 Axes 2 OMag 10000 LSpeed 0 RSpeed 0 LMag 0 RMag 0 LPeriod 60000 RPeriod
18:52:30.091    00006020    Type 0 Axes 2 OMag 10000 LSpeed 13107 RSpeed 0 LMag 2000 RMag 0 LPeriod 60000 RPeriod
18:52:30.259    00006020    Type 0 Axes 2 OMag 10000 LSpeed 0 RSpeed 0 LMag 0 RMag 0 LPeriod 60000 RPeriod
18:52:48.572    00006500    InputHook shutdown
18:52:48.572    00006500    InputHook timed out after 45s

**Comment by Squall-Leonhart** [comment]
Hmm...

Xbox 360 pad with what drivers?
the native windows drivers do not have force feedback.

**Comment by rofled69** [comment]
XBCD 0.27 drivers. Native drivers doesnt work with x360ce.

**Comment by Nucleoprotein** [comment]
Wrong, use pass-trough, but then you cannot reconfigure controls. For reconfiguring Xinput controllers use KrossX Durazno: https://drive.google.com/folderview?id=0B1wssOeJs5_7czZ1WW93SzFxUWs&usp=sharing

**Comment by Squall-Leonhart** [comment]
GTA5's lack of rumble after alt+tab is an issue in gta5 itself

. . . reminds me i have to update xbcd with the rest of the controls i found, controls reported, new rumble library, and signed catalog for win 8 support e.e  ugh..

**Comment by Nucleoprotein** [comment]
GTA5 FFB lose after alt+tab was not related to Timeout ?

**Comment by Squall-Leonhart** [comment]
I believe i tested it with the timeout off,  i'll try it again a bit later, too early in the morning to start up that game xD

**Comment by EvoJocys** [comment]
This report is about the 3.x line, where x360ce was an xinput DLL copied into each game's folder. The current 4.x line works differently: it creates a virtual Xbox 360 controller that every game sees, with nothing copied into game folders. Game-specific problems from the DLL era do not carry over.

The latest release is 4.21 (September 2026): https://github.com/x360ce/x360ce/releases

Closing as part of a tracker clean-up after the 4.21 release. If it still happens on 4.21, reopen this issue with the version and what you see.
